contracts/test/v0.8/CanaryUpkeep.test.ts

```diff
@@ -0,0 +1,222 @@
+import { ethers } from 'hardhat'
+import { BigNumber, Signer } from 'ethers'
+import moment from 'moment'
+import { assert } from 'chai'
+import { CanaryUpkeep } from '../../typechain/CanaryUpkeep'
+import { KeeperRegistry } from '../../typechain/KeeperRegistry'
+import { fastForward, reset } from '../test-helpers/helpers'
+import { getUsers, Personas } from '../test-helpers/setup'
+import { evmRevert } from '../test-helpers/matchers'
+
+let personas: Personas
+let canaryUpkeep: CanaryUpkeep
+let owner: Signer
+let nelly: Signer
+let nancy: Signer
+let ned: Signer
+let keeperAddresses: string[]
+let keeperRegistry: KeeperRegistry
+
+const defaultInterval = 300
+const paymentPremiumPPB = BigNumber.from(250000000)
+const flatFeeMicroLink = BigNumber.from(0)
+const blockCountPerTurn = BigNumber.from(3)
+const stalenessSeconds = BigNumber.from(43820)
+const gasCeilingMultiplier = BigNumber.from(1)
+const checkGasLimit = BigNumber.from(20000000)
+const fallbackGasPrice = BigNumber.from(200)
+const fallbackLinkPrice = BigNumber.from(200000000)
+const maxPerformGas = BigNumber.from(5000000)
+const minUpkeepSpend = BigNumber.from('1000000000000000000')
+const transcoder = ethers.constants.AddressZero
+const registrar = ethers.constants.AddressZero
+const config = {
+  paymentPremiumPPB,
+  flatFeeMicroLink,
+  blockCountPerTurn,
+  checkGasLimit,
+  stalenessSeconds,
+  gasCeilingMultiplier,
+  minUpkeepSpend,
+  maxPerformGas,
+  fallbackGasPrice,
+  fallbackLinkPrice,
+  transcoder,
+  registrar,
+}
+
+before(async () => {
```

Comment:
can you move this under the first `describe` block? otherwise I think it runs before the entire test suite and is not scoped to this test

Reply:
Done